### ISSUE TITLE
Unable to install due to doom-packages-install being undefined

### ISSUE THREAD

**Issue by rpaulo** [doomemacs/doomemacs#7755]
### I confirm that...

- [X] I have searched the [issue tracker](https://github.com/doomemacs/doomemacs/issues), [documentation](https://docs.doomemacs.org), [FAQ](https://docs.doomemacs.org/-/faq), [Discourse](https://discourse.doomemacs.org), and [Google](https://google.com), in case this issue has already been reported/resolved.

- [X] I have read "[How to Debug Issues](https://doomemacs.org/d/how2debug)", and will use it to provide as much information about this issue as possible.

- [X] The issue can be reproduced on the **latest** available commit of Doom.

- [X] The issue can be reproduced on a stable release of Emacs, such as 27, 28, or 29. *(Unstable versions end in .50, .60, or .9x)*

### Expected behavior

I expected Doom Emacs to install.

### Current behavior

Doom crashes:
```
rpaulo@rui-debian-vm ~/.config/emacs % bin/doom install
Installing Doom Emacs!

- Skipping ~/dotfiles/dot.doom.d/ (already exists)
  - Skipping init.el (already exists)
  - Skipping config.el (already exists)
  - Skipping packages.el (already exists)
Generate an envvar file? (see `doom help env` for details) (y or n) y
> Generating envvars file
  ✓ Generated ~/.config/emacs/.local/env
Installing plugins
x There was an unexpected runtime error
  Message: Symbol's function definition is void
  Details: (doom-packages-install)
  Backtrace:
    (doom-packages-install)
    (if (eq install\? :no) (doom-print (doom-print--format (doom-print--class 'warn "Not installing plugins, as requested"))) (doom-print (doom-print--format "Installing plugins")) (doom-packages-insta...
    (let ((default-directory doom-emacs-dir) (yes\? (progn (or (progn (and (memq (type-of context) cl-struct-doom-cli-context-tags) t)) (signal 'wrong-type-argument (list 'doom-cli-context context))) (...
    (let ((config\? (cdr (assq 'config\? alist))) (envfile\? (cdr (assq 'envfile\? alist))) (install\? (cdr (assq 'install\? alist))) (fonts\? (cdr (assq 'fonts\? alist))) (hooks\? (cdr (assq 'hooks\? ...
    ((closure (t) (cli alist) (let ((config\? (cdr (assq 'config\? alist))) (envfile\? (cdr (assq 'envfile\? alist))) (install\? (cdr (assq 'install\? alist))) (fonts\? (cdr (assq 'fonts\? alist))) (ho...
    (funcall (closure (t) (cli alist) (let ((config\? (cdr (assq 'config\? alist))) (envfile\? (cdr (assq 'envfile\? alist))) (install\? (cdr (assq 'install\? alist))) (fonts\? (cdr (assq 'fonts\? alis...
    (doom-cli-execute #s(doom-cli ("doom" "install") nil (("SUMMARY" . "Installs and sets up Doom Emacs for the first time.") ("MAIN" . "") ("This command does the following" . "\n  1. Creates `$DOOMDI...
    (let ((cli x33) (bindings x34)) (doom-cli-execute cli bindings))
    (let* ((x33 (car-safe x32)) (x34 (cdr-safe x32))) (let ((cli x33) (bindings x34)) (doom-cli-execute cli bindings)))
    (progn (ignore (consp x32)) (let* ((x33 (car-safe x32)) (x34 (cdr-safe x32))) (let ((cli x33) (bindings x34)) (doom-cli-execute cli bindings))))
    (let ((x32 (car --dolist-tail--))) (progn (ignore (consp x32)) (let* ((x33 (car-safe x32)) (x34 (cdr-safe x32))) (let ((cli x33) (bindings x34)) (doom-cli-execute cli bindings)))) (setq --dolist-ta...
    (while --dolist-tail-- (let ((x32 (car --dolist-tail--))) (progn (ignore (consp x32)) (let* ((x33 (car-safe x32)) (x34 (cdr-safe x32))) (let ((cli x33) (bindings x34)) (doom-cli-execute cli binding...
  ! Wrote extended backtrace to ~/.config/emacs/.local/state/logs/cli.doom.240324222122.139638.error
```

Let me know if you need the backtrace, but this is easily reproducible on 8d50cd8bfb63e275949eb943fa9d1aad21a5332e on Debian stable.

### Steps to reproduce

Install Doom Emacs on Debian stable

### System Information

system     Debian GNU/Linux 12 (bookworm) Linux 6.1.0-18-amd64 x86_64 emacs      28.2 EMACSDIR=~/.config/emacs/ EMACS=/bin/emacs doom       3.0.0-pre

**Comment by hlissner** [comment]
As of 6d682ee this should be fixed. You'll need to `git pull` to ensure you have the latest version of Doom before running `doom install` again. Thanks for bringing it to my attention!

**Comment by rpaulo** [comment]
@hlissner `doom-profile-env-file-name` is also undefined.  Try `doom sync

**Comment by yaaama** [comment]
I had the same issues, I cleaned out my `~/.config/emacs` folder and cloned in the doom repo and doom install seems ot be working again

**Comment by hlissner** [comment]
@rpaulo Good catch. Fixed in 286be1b.

**Comment by rpaulo** [comment]
Works great now, thanks
